ziti-router/subcmd/run.go

```diff
@@ -105,3 +111,19 @@ func getFlags(cmd *cobra.Command) map[string]*pflag.Flag {
 	})
 	return ret
 }
+
+func waitForShutdown(r *router.Router) {
```

Comment:
I'm pretty sure calling exit here will prevent the tunneler from cleaning up routes and iptables rules on shutdown. See https://github.com/openziti/edge/blob/main/tunnel/intercept/svcpoll.go#L58

Reply:
I feel like signal handlers to shutdown need to be handled by executing cmds and not in the lower deps. The lower deps should expose a `Shutdown()` function and code in the cmds should call those directly or indirectly.

Reply:
the framework should take a `func` as a parameter that the code in the `cmd`s can provide.

Reply:
personally, i think this is kind of a big deal... and definitely a reason to hold up 0.19.8.

Reply:
I agree with setting up the handlers at a higher level closer to the command logic. I also think that calling exit directly in a signal handler function makes it impossible to have more than one signal handler in the process.

Reply:
agreed. and to that end, the framework should support the higher-level calls providing multiple `func`s, and they should all be called at the shutdown event.

Reply:
I don't see a clean way to coordinate multiple separate signal handlers. If there isn't a good way to do that, I don't think we should support it as I think it would be a good idea to try to stop all components, but if we have a misbehaving one, I don't think we should block on it forever. Instead, we should give a certain amount of time, then output a log message and exit. With separate notify handlers, I don't see a clear way to do that.

Reply:
there's one signal handler. that signal handler delegates to one or more functions that are provided to the framework to clean up some part of our runtime. if one of those functions blocks indefinitely, then we shouldn't do anything. in that case, a `kill` will be required to kill the process.

if we end up in cases where we're blocking regularly or often on one of those cleanup functions, then we need to fix that part of our code. but i don't think the framework should try to do anything to mitigate that.

Reply:
Seems reasonable.